### DISCUSSION TITLE
How to use something like SQL's MAX on datetimes?

### DISCUSSION THREAD

**Discussion by zaunermax** [prisma/prisma#3734] [Q&A] [answered]
My usecase is, I want to query events that are connected with orders. But I want to only select the latest events (if existing) for each order.

In pure SQL I'd have a subquery, which selects MAX for the createdAt date. Something like this:

```sql
SELECT *
FROM event
WHERE timestamp = (
    SELECT MAX(timestamp)
    FROM event latest
    WHERE FK_Order = event.FK_Order
)
```

Prisma does not let me aggregate dates, so that was also a dead end. I could not use max with dates either.

The best way I could come up with to achieve my goal was this:

```ts
const events = await this.prisma.event.findMany({
	where: {
		targetId: {
			in: orders.map(({ id }) => id),
		},
	},
	distinct: 'targetId',
	orderBy: {
		updatedAt: 'desc',
	},
});
```

I am not sure though, if this is the best way to achieve my goal, am I missing something? I tried searching stack overflow, the issues discussions and the documentation. Sorry if this is an obvious question.

**Comment by ryands17** [comment, marked as answer]
Hey @zaunermax 👋 

Yes your workaround would be the way to go or a raw query via `prisma.$queryRaw`. Currently Prisma doesn't support aggregating dates, and it would be great if you could create a request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) :)

**Reply by zaunermax** [reply]
Thanks for your answer, prisma is great so far, we adopted it besides typeorm to see if it would hold up and it is very great so far!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
